src/GraphQL.NewtonsoftJson/StringExtensions.cs

```diff
@@ -50,6 +53,38 @@ public static Inputs ToInputs(this JObject obj)
             return GetValueInternal(values) as Dictionary<string, object>;
         }
 
+        /// <summary>
+        /// Deserializes a JSON-formatted string of data into the specified type.
+        /// Any <see cref="Inputs"/> objects will be deserialized into the proper format.
+        /// </summary>
+        public static T FromJson<T>(this string json)
+        {
+            var settings = new JsonSerializerSettings
+            {
+                DateFormatHandling = DateFormatHandling.IsoDateFormat,
+                DateParseHandling = DateParseHandling.None,
+            };
+            settings.Converters.Add(_inputsConverter);
+            return JsonConvert.DeserializeObject<T>(json, settings);
+        }
+
+        /// <summary>
+        /// Deserializes a JSON-formatted stream of data into the specified type.
+        /// Any <see cref="Inputs"/> objects will be deserialized into the proper format.
+        /// </summary>
+        public static async Task<T> FromJsonAsync<T>(this System.IO.Stream stream)
+        {
+            using var streamReader = new System.IO.StreamReader(stream ?? throw new ArgumentNullException(nameof(stream)), System.Text.Encoding.UTF8, false, 1024, true);
+            var text = await streamReader.ReadToEndAsync();
```

Comment:
It defeats the purpose of stream. See https://www.newtonsoft.com/json/help/html/DeserializeWithJsonSerializerFromFile.htm and use StreamReader instead with JsonSerializer.

Reply:
Well, it does read asynchronously, at the cost of buffering the whole input.  I made the suggested change and changed the method to a synchronous method here.